<center>
    <img src="https://i.ibb.co/wQg0qzq/Banner.png" width="900" alt="KCode Banner"  />
</center>
<a href="https://www.linkedin.com/in/kauebr/"><img src="https://cdn-icons-png.flaticon.com/512/174/174857.png" width="30" ></a> <a href="mailto: kauemandarino@gmail.com"><img src="https://cdn-icons-png.flaticon.com/512/5968/5968534.png" width="30" ></a><a href="
https://api.whatsapp.com/send?phone=5541991835800&text=Oi"> <img src="https://imagepng.org/wp-content/uploads/2017/08/whatsapp-icone-1.png" width="30" ></a><a href="https://github.com/kauebr/"><img src="https://github.githubassets.com/images/modules/logos_page/Octocat.png" width="30" ></a>

<HR>

> AINDA EM CRIAÇÃO... 

# Regressão linear múltipla
        Case de clientes de um banco, retirado do Kaggle em que a instituição estava querendo entender o motivo de uma alta no índice de cancelamentos
<HR>

## Objetivos:
*   Tratar paralelos e entender  possívels causas dos cancelamentos;
*   Gerar insights que possam ajudar na resolução do problema;
*   Usar a biblioteca scikit-learn para implantar um modelo de Regressão Linear Múltiplo;   
*   Criar, treinar e testar o modelo;
<HR>


<h1>Índice</h1>

<div class="alert alert-block alert-info" style="margin-top: 20px">
    <ol>
        <li><a href=#DF>Criando DataFrame</a></li>
        <li><a href=#Dados>Entendendo os dados</a></li>
        <li><a href=#Graf>Gráficos</a></li>
        <li><a href=#TT>Treino e teste</a></li>
        <li><a href=#Reg>Criando Modelo de regressão linear múltipla</a></li>
        <li><a href=#Pred>Predições e Avaliações</a></li>    
    </ol>
</div>
<br>
<hr>



### Importando bibliotecas e recursos pertinentes

In [4]:
import matplotlib.pyplot as plt
import pandas as pd
import pylab as pl
import numpy as np
%matplotlib inline

# <h2 id="DF">Criando DataFrame</h2>

In [11]:
df = pd.read_csv('ClientesBanco.csv', encoding='latin1')
df.head(5).T

0                1                2  \
CLIENTNUM                         768805383        818770008        713982108   
Categoria                           Cliente          Cliente          Cliente   
Idade                                    45               49               51   
Sexo                                      M                F                M   
Dependentes                               3                5                3   
Educação                       Ensino Médio  Ensino Superior  Ensino Superior   
Estado Civil                         Casado         Solteiro           Casado   
Faixa Salarial Anual            $60K - $80K   Less than $40K     $80K - $120K   
Categoria Cartão                       Blue             Blue             Blue   
Meses como Cliente                       39               44               36   
Produtos Contratados                      5                6                4   
Inatividade 12m                           1                1                1   
Contatos 12m                              3                2                0   
Limite                              12691.0           8256.0           3418.0   
Limite Consumido                        777              864                0   
Limite Disponível                   11914.0           7392.0           3418.0   
Mudanças Transacoes_Q4_Q1            1335.0           1541.0           2594.0   
Valor Transacoes 12m                   1144             1291             1887   
Qtde Transacoes 12m                      42               33               20   
Mudança Qtde Transações_Q4_Q1        1625.0           3714.0           2333.0   
Taxa de Utilização Cartão             0.061            0.105              0.0   

                                            3                  4  
CLIENTNUM                           769911858          709106358  
Categoria                             Cliente            Cliente  
Idade                                      40                 40  
Sexo                                        F                  M  
Dependentes                                 4                  3  
Educação                         Ensino Médio  Sem ensino formal  
Estado Civil                    Não informado             Casado  
Faixa Salarial Anual           Less than $40K        $60K - $80K  
Categoria Cartão                         Blue               Blue  
Meses como Cliente                         34                 21  
Produtos Contratados                        3                  5  
Inatividade 12m                             4                  1  
Contatos 12m                                1                  0  
Limite                                 3313.0             4716.0  
Limite Consumido                         2517                  0  
Limite Disponível                       796.0             4716.0  
Mudanças Transacoes_Q4_Q1              1405.0             2175.0  
Valor Transacoes 12m                     1171                816  
Qtde Transacoes 12m                        20                 28  
Mudança Qtde Transações_Q4_Q1          2333.0                2.5  
Taxa de Utilização Cartão                0.76                0.0

## <h2 id="Dados">Entendendo os dados</h2>


### `Clientes_Banco.csv`:


Conjunto de dados **`Clientes_Banco.csv`**, Case Real, de um conjunto de dados já traduzido, referentes à clientes cartão de crédito de um banco X.


*   **CLIENTENUM** - 123456789 - **Número de identificação do cliente**
*   **Categoria** - Cliente/Cancelado - **Status de atividade**
*   **Idade** - 45 - **Idade do cliente**
*   **Sexo** - M/F
*   **Dependentes** - 3 - **Número de dependentes do cliente**
*   **Educação** - Ensino médio - **Grau de escolaridade do s**
*   **Estado Civil** - Solteiro
*   **Faixa Salarial Anual** - **Salário Anual**
*   **Categoria Cartão'** - Blue - **Categoria do cartão do cliente**
*   **Meses como Cliente** - 39 - **Há quanto tempo consta na base de clientes**
*   **Produtos Contratados** - 5 - **Quantos produtos tem contratado**
*   **Inatividade 12m** - 1 - **Quanto Tempo cliente passou sem utilizar** 
*   **Contatos 12m** - 3 - **Contatos que o cliente fez no ultimo ano**
*   **Limite** - 12691.0 - **Limite total**
*   **Limite Consumido** - 777 - **Limite utilizado**
*   **Limite Disponível** - 11914.0 - **Limite disponível**
*   **Mudanças Transacoes_Q4_Q1** - 1335.0 - **Valor de mudanças nas transações do quarto quadrimestre ao primeiro**
*   **Valor Transacoes 12m** - 1144 - **Valor total das transações**
*   **Mudança Qtde Transações_Q4_Q1** - 1625.0 - **Mudanças na quantidade de transações do primeiro ao quarto quadrimestre**
*   **Taxa de Utilização Cartão** - 0.061 - **Taxa de utilização de 0 a 1, aonde 1 é maior**
 


Vamos agora selecionar umas características que usaremos para a regressão

## <h2 id="Graf">Gráficos</h2>

Agora vamos plotar gráficos para ter uma noção visual de se as relações entre as características dos veículos e a emissão de CO2 são lineares, pois só podemos usar este modelo de `Regressão linear múltipla` se as releções forem lineares:

> Continuarei em breve... 